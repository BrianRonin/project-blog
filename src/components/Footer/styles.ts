import styled, { css } from 'styled-components'
import { Main as Text } from '../Text/styles'

export const Main = styled.div`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    background-color: ${theme.colors.white};
    a {
      color: ${theme.colors.darkText};
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    & ${Text} {
      font-size: ${theme.fonts.sizes.small};
    }
  `}
`
