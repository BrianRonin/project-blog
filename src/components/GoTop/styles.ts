import styled, { css, DefaultTheme } from 'styled-components'

const Light = (theme: DefaultTheme) => css`
  color: ${theme.colors.primary};
  background: ${theme.colors.white};
`
type background = {
  background: boolean
}

export const Main = styled.a<background>`
  ${({ theme, background }) => css`
    position: fixed;
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.8;
    border-radius: 50%;
    ${background && Light(theme)};
  `}
`
