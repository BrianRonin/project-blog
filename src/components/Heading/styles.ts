import styled, { css, DefaultTheme } from 'styled-components'
import { HeadingProps } from '.'

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.large};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xlarge};
    @media ${theme.media.medium} {
      font-size: ${theme.sizes.large};
    }
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xhuge};
    ${mediaFont(theme)}
  `,
}

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.sizes.xlarge};
  }
`

export const Title = styled.h1<
  Pick<HeadingProps, 'light' | 'size' | 'uppercase'>
>`
  ${({ theme, light, size, uppercase }) => css`
    color: ${light ? theme.colors.white : theme.colors.primary};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
    ${!!size && titleSize[size](theme)};
  `}
`
