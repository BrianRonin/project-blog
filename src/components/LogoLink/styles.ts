import styled from 'styled-components'
export const Main = styled.a`
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;

  > img {
    width: 7rem;

    border-radius: 50%;
  }
`

export const Logo = styled.img`
  max-height: 6rem;
`
