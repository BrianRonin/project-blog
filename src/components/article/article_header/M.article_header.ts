import { mock_header } from '../../header/M.header'
import { mock_cover } from '../cover/M.cover'
import { articleHeaderProps } from './article_header'

export const mock_article_header = {
  title: 'titulo qualquer',
  excerpt: 'uma descrição que vou criando enquanto penso na descrição',
  coverProps: {
    cover: {
      url: 'https://miro.medium.com/max/640/0*i1v1In2Tn4Stnwnl.jpg',
      altText: 'um alt image',
    },
    articleMetaProps: {
      category: {
        name: 'Tecnologia',
        slug: 'tecnologia',
      },
      createdAt: '2022-10-05T02:48:15.628Z',
      tags: [
        {
          name: 'React',
          slug: 'react',
        },
        {
          name: 'Docker',
          slug: 'docker',
        },
      ],
      author: {
        name: 'Brian',
        slug: 'brian',
      },
    },
  },
} as articleHeaderProps
