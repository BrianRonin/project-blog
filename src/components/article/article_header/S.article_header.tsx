import styled, { css, DefaultTheme } from 'styled-components'
import { Title as heading } from '../../../components/Heading/styles'

const name = (theme: DefaultTheme) => css`
  //
`

export const Main = styled.div`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${heading} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`
export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.spacings.medium};
    margin: ${theme.spacings.medium} 0;
    @media ${theme.media.medium} {
      font-size: ${theme.spacings.small};
    }
  `}
`
