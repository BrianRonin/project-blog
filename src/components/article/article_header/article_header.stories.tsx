import { Meta, Story } from '@storybook/react/types-6-0'
import { ArticleHeader, articleHeaderProps } from './article_header'
import { mock_article_header } from './M.article_header'

export default {
  title: 'Article header',
  component: ArticleHeader,
  args: mock_article_header,
  argTypes: {
    //
  },
} as Meta

export const Template: Story<articleHeaderProps> = (args) => (
  <>
    <ArticleHeader {...args} />
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Laborum deleniti
      odit deserunt non consectetur voluptatem officiis quae totam. Architecto,
      quia debitis? Reiciendis quas doloribus ab esse nemo sint voluptatibus
      commodi?
    </p>
  </>
)

Template.parameters = {
  backgrounds: {
    default: 'light',
  },
}
