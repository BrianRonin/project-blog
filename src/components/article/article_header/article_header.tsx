import {
  ArticleMeta,
  articleMetaProps,
} from '../article_meta/article_meta'
import {
  CoverWithMeta,
  coverWithMetaProps,
} from '../cover_with_meta/cover_with_meta'
import * as S from './S.article_header'
import {
  Heading,
  HeadingProps,
} from '../../Heading/index'
import { Header } from '../../header/header'
import { headerProps } from '../../header/header'

export type articleHeaderProps = {
  title: string
  excerpt: string
  coverWithMetaProps: coverWithMetaProps
}

export const ArticleHeader = ({
  coverWithMetaProps,
  excerpt,
  title,
}: articleHeaderProps) => {
  return (
    <S.Main>
      <Heading size='big'>{title}</Heading>
      <S.Excerpt>{excerpt}</S.Excerpt>
      <CoverWithMeta {...coverWithMetaProps} />
    </S.Main>
  )
}
