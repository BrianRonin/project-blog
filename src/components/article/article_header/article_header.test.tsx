import { screen } from '@testing-library/react'
import { renderTheme } from '~/styles/render-theme'
import { ArticleHeader } from './article_header'
import { mock_article_header } from './M.article_header'
describe('<ArticleHeader />', () => {
  it('should render', () => {
    const { container, debug } = renderTheme(
      <ArticleHeader {...mock_article_header} />,
    )
    expect(screen.getByText(mock_article_header.excerpt)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
    debug()
  })
})
