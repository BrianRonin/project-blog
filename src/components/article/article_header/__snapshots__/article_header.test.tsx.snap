// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleHeader /> should render 1`] = `
.c4 {
  width: 100%;
  max-width: 96rem;
  margin-bottom: 2.4rem;
}

.c5 {
  font-size: 2.4rem;
  color: #4f4f4fff;
  font-style: italic;
}

.c5 .tags span::after {
  content: ', ';
}

.c5 .tags span:last-child::after {
  content: '';
}

.c5 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300msa ease-in-out;
  transition: all 300msa ease-in-out;
}

.c5 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c2 {
  color: #000000;
  text-transform: none;
  font-size: 4.0rem;
}

.c0 {
  padding-bottom: 4.0rem;
  margin-bottom: 4.0rem;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c0 .c1 {
  margin: 0;
  margin-bottom: 2.4rem;
}

.c3 {
  font-size: 2.4rem;
  margin: 2.4rem 0;
}

@media (max-width:768px) {
  .c5 {
    font-size: 1.6rem;
  }
}

@media (max-width:768px) {
  .c2 {
    font-size: 3.2rem;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 1.6rem;
  }
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      titulo qualquer
    </h1>
    <p
      class="c3"
    >
      uma descrição que vou criando enquanto penso na descrição
    </p>
    <div
      class=""
    >
      <div>
        <img
          alt="um alt image"
          class="c4"
          src="https://miro.medium.com/max/640/0*i1v1In2Tn4Stnwnl.jpg"
        />
        <em>
          <div
            class="c5 id-component-article-meta"
          >
            <span
              class=""
            >
              <p>
                Por
                 
                <a
                  href="/author/brian"
                >
                  <strong>
                    Brian
                  </strong>
                </a>
                 
                em 
                <time
                  datetime="2022-10-05T02:48:15.628Z"
                >
                  4 de out. de 2022
                  , 
                </time>
                <a
                  href="/category/tecnologia"
                >
                  <strong>
                    Tecnologia, 
                  </strong>
                </a>
                <span
                  class="tags"
                >
                  <span>
                    <a
                      href="/tags/react"
                    >
                      React
                    </a>
                  </span>
                  <span>
                    <a
                      href="/tags/docker"
                    >
                      Docker
                    </a>
                  </span>
                </span>
              </p>
            </span>
          </div>
        </em>
      </div>
    </div>
  </div>
</div>
`;
